slo-monitor/src/monitors/store.go

```diff
@@ -0,0 +1,196 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package monitors
+
+import (
+	"sync"
+
+	"k8s.io/client-go/tools/cache"
+
+	"github.com/golang/glog"
+)
+
+const (
+	updateOperationType = "update"
+	deleteOperationType = "delete"
+)
+
+type workItem struct {
+	operationType string
+	key           string
+	obj           interface{}
+}
+
+// NoStoreQueue implements k8s.io/client-go/tools/cache Store and Queue interfaces
+type NoStoreQueue struct {
+	sync.Mutex
+
+	cond      sync.Cond
+	queue     []workItem
+	indices   map[string]int64
+	nextIndex int64
+	offset    int64
```

Comment:
Please add components to those fields - before looking into implementation, I have no idea what these are.